shuup/front/utils/product.py

```diff
@@ -114,7 +117,7 @@ def _unpack_orderable_variation_children_from_cache(cached_value):
     return (orderable_variation_children, orderable)
 
 
-def get_orderable_variation_children(product, request, variation_variables):    # noqa (C901)
+def get_orderable_variation_children(product, request, variation_variables, supplier=None):    # noqa (C901)
```

Comment:
I guess you can pass this supplier in the context cache fetch below to prevent returning wrong values when supplier changes

Reply:
Thats not safe I think. Also this method is not open to world in reality (only place this is called is there at line 62. Maybe that shouldn't be optional but just in case.

Reply:
I mean, check line 124 below: `key, val = context_cache.get_cached_value(` here you don't pass the supplier, this way a different supplier can make the cache fetch a wrong value (from a different supplier).

Reply:
You were correct :) Tests now covers this.